python/runtime/model/model.py

```diff
@@ -67,56 +69,132 @@ def __init__(self, typ, meta):
+            local_dir (str): the local directory to save.
 
-    Args:
-        datasource: string
-            The connection string to DBMS
+        Returns:
+            None.
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tarball = os.path.join(tmp_dir, TARBALL_NAME)
+            self._zip(local_dir, tarball)
+
+            def _bytes_reader(filename, buf_size=8 * 32):
+                def _gen():
+                    with open(filename, "rb") as f:
+                        while True:
+                            data = f.read(buf_size)
+                            if data:
+                                yield data
+                            else:
+                                break
+
+                return _gen
+
+            write_with_generator(datasource, table, _bytes_reader(tarball))
+
+    @staticmethod
+    def load_from_db(datasource, table, local_dir=os.getcwd()):
+        """
+        Load the saved model from DBMS and unzip it on local_dir.
 
-        table: string
-            The table name which saved in DBMS
+        Args:
+            datasource (str): the connection string to DBMS
+            table (str): the table name which saved in DBMS
+            local_dir (str): the local directory to load.
 
-    Returns:
-        Model: a Model object represent the model type and meta information.
-    """
-    gen = read_with_generator(datasource, table)
-    with open(tarball, "wb") as f:
-        for data in gen():
-            f.write(bytes(data))
+        Returns:
+            Model: a Model object represent the model type and meta
+            information.
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tarball = os.path.join(tmp_dir, TARBALL_NAME)
+            gen = read_with_generator(datasource, table)
+            with open(tarball, "wb") as f:
+                for data in gen():
+                    f.write(bytes(data))
 
-    unzip_dir(tarball, cwd)
-    return _load_pkl(os.path.join(cwd, model_obj_file))
+            return Model._unzip(local_dir, tarball)
+
+    def save_to_oss(self, oss_model_dir, local_dir=os.getcwd()):
+        """
+        This save function would archive all the files on local_dir
+        into a tarball, and save it into OSS model directory.
+
+        Args:
+            oss_model_dir (str): the OSS model directory to save.
+            local_dir (str): the local directory to save.
+
+        Returns:
+            None.
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tarball = os.path.join(tmp_dir, TARBALL_NAME)
+            self._zip(local_dir, tarball)
+            oss.save_file(oss_model_dir, tarball, TARBALL_NAME)
+
+    @staticmethod
+    def load_from_oss(oss_model_dir, local_dir=os.getcwd()):
+        """
+        Load the saved model from OSS and unzip it on local_dir.
+
+        Args:
+            oss_model_dir (str): the OSS model directory to load.
```

Comment:
Maybe document the format of `oss_model_dir` should be like `oss://bucket/path/to/dir`?

Reply:
Done.